flang/lib/Optimizer/Transforms/OMPEarlyOutlining.cpp

```diff
@@ -99,6 +100,18 @@ class OMPEarlyOutliningPass
       return;
     }
 
+    if (auto declareOp = mlir::dyn_cast_if_present<hlfir::DeclareOp>(
```

Comment:
Nit: Add a comment for this block of code.

Reply:
Done.